platform/ecs/task/task.go

```diff
@@ -204,9 +205,21 @@ func parseTaskArn(taskArn string) (string, error) {
 
 func (t *task) getResourceLimits() (ResourceLimits, error) {
 	var limits = ResourceLimits{}
-	subgroup := filepath.Join(ecsSubgroupName, t.id)
 
-	cgCPU, err := t.cgroup.CPU(subgroup)
+	cg, err := t.proc.Cgroup()
+	if err != nil {
+		return limits, err
+	}
+
+	memSs, ok := cg[memorySubsystem]
+	if !ok {
+		return limits, fmt.Errorf("%s subsystem not exists", memorySubsystem)
+	}
+
+	taskSG, _ := path.Split(memSs.CgroupPath) // CgroupPath: "/ecs/TASK_ID/DOCKER_ID" or "/ecs/CLUSTER_NAME/TASK_ID/DOCKER_ID"
+	sg := path.Clean(taskSG)                  // remove tailing slash
```

Comment:
The logic is long enough to define a new function, func (t *task) getTaskSubGroup for example.